File: backend/app/schemas/skill.py
"""技能相关模式"""
from pydantic import BaseModel, Field
from typing import Optional
from datetime import datetime


class SkillBase(BaseModel):
    """技能基础模式"""
    name: str = Field(..., min_length=1, max_length=100, description="技能名称")
    proficiency: str = Field(..., description="熟练度: familiar(熟悉), proficient(熟练), expert(精通)")


class SkillCreate(SkillBase):
    """创建技能请求"""
    pass


class SkillUpdate(BaseModel):
    """更新技能请求"""
    name: Optional[str] = Field(None, min_length=1, max_length=100, description="技能名称")
    proficiency: Optional[str] = Field(None, description="熟练度")


class SkillResponse(BaseModel):
    """技能响应"""
    id: int
    user_id: int
    name: str
    proficiency: str
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class SkillListResponse(BaseModel):
    """技能列表响应"""
    total: int
    items: list[SkillResponse]

File: frontend/src/api/skill.ts
/**
 * 技能管理API
 */
import request from './index'

export interface Skill {
  id: number
  user_id: number
  name: string
  proficiency: string
  created_at: string
  updated_at: string
}

export interface SkillCreate {
  name: string
  proficiency: 'familiar' | 'proficient' | 'expert'
}

export interface SkillUpdate {
  name?: string
  proficiency?: 'familiar' | 'proficient' | 'expert'
}

export interface SkillListResponse {
  total: number
  items: Skill[]
}

/**
 * 创建技能
 */
export function createSkill(data: SkillCreate): Promise<Skill> {
  return request.post('/api/v1/skills', data)
}

/**
 * 获取技能列表
 */
export function getSkills(): Promise<SkillListResponse> {
  return request.get('/api/v1/skills')
}

/**
 * 获取技能详情
 */
export function getSkill(skillId: number): Promise<Skill> {
  return request.get(`/api/v1/skills/${skillId}`)
}

/**
 * 更新技能
 */
export function updateSkill(skillId: number, data: SkillUpdate): Promise<Skill> {
  return request.put(`/api/v1/skills/${skillId}`, data)
}

/**
 * 删除技能
 */
export function deleteSkill(skillId: number): Promise<void> {
  return request.delete(`/api/v1/skills/${skillId}`)
}
